common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java

```diff
@@ -304,7 +304,7 @@ public static void throwException(Throwable t) {
   static {
     boolean _unaligned;
     String arch = System.getProperty("os.arch", "");
-    if (arch.equals("ppc64le") || arch.equals("ppc64")) {
+    if (arch.equals("ppc64le") || arch.equals("ppc64") || arch.contains("s390x")) {
```

Comment:
Will it create false positive case if we compare using `contains()` ?

Reply:
@ketank-new it's worth asking, what arch strings are you matching here? i.e. according to https://www.ibm.com/support/knowledgecenter/en/SSYKE2_8.0.0/com.ibm.java.80.doc/diag/preface/conventions.html do you also need to match "s390"?

Reply:
we have tested our changes on 64 bit 'Z ' machine and that is what we are currently supporting so we just want to check for s390x  only and not s390

Reply:
Is "s390x" the only string that should trigger this? then use `.equals`. Or do these machines report variations on "s390x" for which this is a substring?

Reply:
we think "s390x" is good , no variations in string reported across "Z" machines
do u want me to change "matches" to ".equals" now?

Reply:
Yes, if you're just looking for 's390x', change it to `.equals`. Otherwise looking good.